Tests/syntax_test_css.scss

```diff
@@ -0,0 +1,4096 @@
+/*                   ^^ meta.number.integer.decimal.css constant.numeric.value.css */
+
+.test:nth-child(-n+ 3) {}
+/*              ^ keyword.operator.arithmetic */
+/*               ^ meta.number.integer.decimal.css constant.numeric.suffix.css */
+/*                ^ keyword.operator.arithmetic */
+/*                 ^^ meta.number.integer.decimal.css constant.numeric.value.css */
+
+.test:nth-child(-n +3) {}
+/*              ^ keyword.operator.arithmetic */
+/*               ^ meta.number.integer.decimal.css constant.numeric.suffix.css */
+/*                ^ - constant */
+/*                 ^ keyword.operator.arithmetic */
+/*                  ^ meta.number.integer.decimal.css constant.numeric.value.css */
+
+.test:nth-child(+3) {}
+/*              ^ keyword.operator.arithmetic */
+/*               ^ meta.number.integer.decimal.css constant.numeric.value.css */
+
+.test-pseudo-classes:nth-child(2):hover {}
+/*^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ meta.selector.css */
+/*                                      ^^ - meta.selector.css */
+/*                   ^^^^^^^^^ entity.other.pseudo-class.css */
+/*                             ^ meta.number.integer.decimal.css constant.numeric.value.css */
+/*                               ^ punctuation.definition.pseudo-class.css - entity */
+/*                                ^^^^^ entity.other.pseudo-class.css */
+
+.test-pseudo-class-numerics:nth-last-of-type(-n+3) {}
+/*                         ^ punctuation.definition.pseudo-class.css - entity */
+/*                          ^^^^^^^^^^^^^^^^ entity.other.pseudo-class.css */
+/*                                           ^ keyword.operator.arithmetic */
+/*                                            ^ meta.number.integer.decimal.css constant.numeric.suffix.css */
+/*                                             ^ keyword.operator.arithmetic */
+/*                                              ^ meta.number.integer.decimal.css constant.numeric.value.css */
+
+.test-pseudo-class-keywords:nth-of-type(odd) {}
+/*                         ^ punctuation.definition.pseudo-class.css - entity */
+/*                          ^^^^^^^^^^^ entity.other.pseudo-class.css */
+/*                                      ^^^ support.constant.property-value.css */
+
+.test-pseudo-class-strings:dir {}
+/*                        ^ punctuation.definition.pseudo-class.css - entity */
+/*                         ^^^ entity.other.pseudo-class.css - meta.function-call */
+
+.test-pseudo-class-strings:dir(ltr) {}
+/*                         ^^^ meta.function-call.identifier.css */
+/*                            ^^^^^ meta.function-call.arguments.css meta.group.css */
+/*                        ^ punctuation.definition.pseudo-class.css - entity */
+/*                         ^^^ entity.other.pseudo-class.css */
+/*                             ^^^ support.constant.property-value.css */
+
+// TODO: unsure what to to about *- etc in unquoted strings
```

Comment:
@deathaxe I did disable a number of test, with TODO comments. If you have ideas about those, that would be very welcome. 

In particular, I can't seem to wrap my head around nested properties in SCSS: https://github.com/braver/Sass/pull/96/files#diff-80093a5df0f618ca72f6a76e542c74d5124295ef431caabf8fe1df0cd3876e7bR213

Reply:
I'll walk through the files.

The link seems to point to a rather huge diff.

Reply:
The diff's not really important actually, I rewrote all tests from scratch anyway. .. yeah so that link is useless, sorry. Have a look at this section of the test: https://github.com/braver/Sass/blob/simplify/Tests/syntax_test_scss.scss#L211-L211. I did not actually add tests there. `transition: blue {` is seen as a selector, but it's actually a property. I think this pattern of `word-colon-space-word-space-curlybrace` should somehow match for `maybe-property`. ..